### PR TITLE
Bump backport-action to v1 major release

### PR DESCRIPTION
**Pull request by korthout**
### Description of your changes

This pull requests bumps the backport-action to its latest version, and adjusts the workflow accordingly.

The `v1` release of this backport-action has a few improvements:
- faster, it no longer requires fetch-depth 0 checkouts and it fetches only the necessary commits
- major release tag moves along with patches (SemVer)
- it no longer requires any inputs by default

Note that the repo moved from zeebe-io to korthout. See [releases notes v1.0.0](https://github.com/korthout/backport-action/releases/tag/v1.0.0).

I have:

- [x] Read and followed Crossplane's [contribution process].
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

- Zeebe itself has switched to this version https://github.com/camunda/zeebe/blob/main/.github/workflows/backport.yml#L30
- The action is tested using https://github.com/korthout/backport-action-test

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by korthout** [comment]
@negz Could you have a look at this?

Please note that I was unable to run `make reviewable`: 

```
 make reviewable
make: *** No rule to make target `reviewable'.  Stop.
```

**Comment by ytsarev** [comment]
@korthout please run `make submodules`, it should make `make reviewable` available :)

**Comment by korthout** [comment]
Thanks @ytsarev 

I ran into another issue. Could you help me out?

```
make reviewable
10:47:40 [ .. ] verify dependencies have expected content
all modules verified
10:47:42 [ OK ] go modules dependencies verified
10:47:42 [ .. ] go generate darwin_amd64
10:47:48 [ OK ] go generate darwin_amd64
10:47:48 [ .. ] go mod tidy
10:47:48 [ OK ] go mod tidy
10:47:48 [ .. ] verify dependencies have expected content
all modules verified
10:47:50 [ OK ] go modules dependencies verified
10:47:51 [ .. ] golangci-lint
WARN [runner] Can't run linter goanalysis_metalinter: bodyclose: failed prerequisites: [buildssa@github.com/crossplane/provider-template/apis/sample/v1alpha1: analysis skipped: errors in package: [/Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/groupversion_info.go:24:2: could not import k8s.io/apimachinery/pkg/runtime/schema (/Users/korthout/dev/github/provider-template/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.25.3/pkg/runtime/schema/generated.pb.go:23:6: could not import fmt (/usr/local/Cellar/go/1.19.4/libexec/src/fmt/errors.go:7:8: could not import errors (/usr/local/Cellar/go/1.19.4/libexec/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/Cellar/go/1.19.4/libexec/src/internal/reflectlite/swapper.go:8:2: could not import internal/goarch (-: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool))))) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/groupversion_info.go:25:2: could not import sigs.k8s.io/controller-runtime/pkg/scheme (/Users/korthout/dev/github/provider-template/.work/pkg/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.0/pkg/scheme/scheme.go:58:9: could not import k8s.io/apimachinery/pkg/apis/meta/v1 (/Users/korthout/dev/github/provider-template/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.25.3/pkg/apis/meta/v1/controller_ref.go:20:2: could not import k8s.io/apimachinery/pkg/runtime/schema (/Users/korthout/dev/github/provider-template/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.25.3/pkg/runtime/schema/generated.pb.go:23:6: could not import fmt (/usr/local/Cellar/go/1.19.4/libexec/src/fmt/errors.go:7:8: could not import errors (/usr/local/Cellar/go/1.19.4/libexec/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/Cellar/go/1.19.4/libexec/src/internal/reflectlite/swapper.go:8:2: could not import internal/goarch (-: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool))))))) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/mytype_types.go:20:2: could not import reflect (/usr/local/Cellar/go/1.19.4/libexec/src/reflect/abi.go:8:2: could not import internal/abi (/usr/local/Cellar/go/1.19.4/libexec/src/internal/abi/abi.go:8:2: could not import internal/goarch (-: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool))) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/mytype_types.go:22:9: could not import k8s.io/apimachinery/pkg/apis/meta/v1 (/Users/korthout/dev/github/provider-template/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.25.3/pkg/apis/meta/v1/controller_ref.go:20:2: could not import k8s.io/apimachinery/pkg/runtime/schema (/Users/korthout/dev/github/provider-template/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.25.3/pkg/runtime/schema/generated.pb.go:23:6: could not import fmt (/usr/local/Cellar/go/1.19.4/libexec/src/fmt/errors.go:7:8: could not import errors (/usr/local/Cellar/go/1.19.4/libexec/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/Cellar/go/1.19.4/libexec/src/internal/reflectlite/swapper.go:8:2: could not import internal/goarch (-: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool)))))) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/mytype_types.go:25:7: could not import github.com/crossplane/crossplane-runtime/apis/common/v1 (/Users/korthout/dev/github/provider-template/.work/pkg/pkg/mod/github.com/crossplane/crossplane-runtime@v0.18.0/apis/common/v1/condition.go:20:2: could not import sort (/usr/local/Cellar/go/1.19.4/libexec/src/sort/slice.go:7:8: could not import math/bits (-: could not load export data: cannot import "math/bits" (unknown iexport format version 2), export data is newer version - update tool))) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/zz_generated.deepcopy.go:25:10: could not import k8s.io/apimachinery/pkg/runtime (/Users/korthout/dev/github/provider-template/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.25.3/pkg/runtime/allocator.go:20:2: could not import sync (/usr/local/Cellar/go/1.19.4/libexec/src/sync/cond.go:8:2: could not import sync/atomic (/usr/local/Cellar/go/1.19.4/libexec/src/sync/atomic/type.go:39:16: expected ']', found any))) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/zz_generated.managedlist.go:20:17: could not import github.com/crossplane/crossplane-runtime/pkg/resource (/Users/korthout/dev/github/provider-template/.work/pkg/pkg/mod/github.com/crossplane/crossplane-runtime@v0.18.0/pkg/resource/api.go:20:2: could not import context (/usr/local/Cellar/go/1.19.4/libexec/src/context/context.go:51:2: could not import errors (/usr/local/Cellar/go/1.19.4/libexec/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/Cellar/go/1.19.4/libexec/src/internal/reflectlite/swapper.go:8:2: could not import internal/goarch (-: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool))))) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/zz_generated.managed.go:24:19: mg.Status.GetCondition undefined (type MyTypeStatus has no field or method GetCondition) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/zz_generated.managed.go:29:17: mg.Spec.DeletionPolicy undefined (type MyTypeSpec has no field or method DeletionPolicy) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/zz_generated.managed.go:34:17: mg.Spec.ProviderConfigReference undefined (type MyTypeSpec has no field or method ProviderConfigReference) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/zz_generated.managed.go:42:17: mg.Spec.ProviderReference undefined (type MyTypeSpec has no field or method ProviderReference) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/zz_generated.managed.go:47:17: mg.Spec.PublishConnectionDetailsTo undefined (type MyTypeSpec has no field or method PublishConnectionDetailsTo) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/zz_generated.managed.go:52:17: mg.Spec.WriteConnectionSecretToReference undefined (type MyTypeSpec has no field or method WriteConnectionSecretToReference) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/zz_generated.managed.go:57:12: mg.Status.SetConditions undefined (type MyTypeStatus has no field or method SetConditions) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/zz_generated.managed.go:62:10: mg.Spec.DeletionPolicy undefined (type MyTypeSpec has no field or method DeletionPolicy) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/zz_generated.managed.go:67:10: mg.Spec.ProviderConfigReference undefined (type MyTypeSpec has no field or method ProviderConfigReference) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/zz_generated.managed.go:75:10: mg.Spec.ProviderReference undefined (type MyTypeSpec has no field or method ProviderReference) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/zz_generated.managed.go:80:10: mg.Spec.PublishConnectionDetailsTo undefined (type MyTypeSpec has no field or method PublishConnectionDetailsTo) /Users/korthout/dev/github/provider-template/apis/sample/v1alpha1/zz_generated.managed.go:85:10: mg.Spec.WriteConnectionSecretToReference undefined (type MyTypeSpec has no field or method WriteConnectionSecretToReference)]]
WARN [runner] Can't run linter unused: buildir: failed to load package goarch: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool
ERRO Running error: buildir: failed to load package goarch: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool
10:47:55 [FAIL]
make[2]: *** [go.lint] Error 1
make[1]: *** [lint] Error 2
make: *** [reviewable] Error 2
```

I'm not so familiar with Go. It seems there's an issue with a version. I'm on go version go1.19.4 darwin/amd64.

File: .github/workflows/commands.yml
Comment (ytsarev) [inline .github/workflows/commands.yml:86]: do you want to replace this one as well?

Comment (korthout) [reply .github/workflows/commands.yml:86]: Good find! Yes I'll adjust it immediately

**Comment by korthout** [comment]
@ytsarev It might just be a problem on my machine. Could you please run the [CI workflow](https://github.com/crossplane/provider-template/blob/main/.github/workflows/ci.yml) on these changes? As I read that file, it should also perform the lining (although with a different tool).

**Comment by ytsarev** [comment]
@korthout looks like I don't have access rights to this repo, I'll ask around

**Review by jbw976 — COMMENTED** [review]
Thanks for this effort @korthout!! Catching up a bit here, I see that you are the author behind this backport action, thanks for making this tool that we have found useful!

I verified the repo movement to your korthout account, so this looks good to approve the workflow changes and run it! Let's see what CI says 🙏

**Comment by korthout** [comment]
Thanks @jbw976, glad you like it. I think this is only waiting on an approving review. The CI checks have been passed successfully.

**Comment by korthout** [comment]
@jbw976 Is there anything I should do before this can be approved?

**Review by jbw976 — APPROVED** [review]
Looks like CI is green here, thanks for the reminder to look at this PR again @korthout!  Really appreciate your proactive action here to get these versions updated.  Keep up the great work! 🙇‍♂️

**Pull request merged**
